services/web/server/src/simcore_service_webserver/socketio/server.py

```diff
@@ -3,55 +3,71 @@
 from typing import AsyncIterator
 
 from aiohttp import web
-from socketio import AsyncServer
+from socketio import AsyncAioPikaManager, AsyncServer
 
-from ._utils import APP_CLIENT_SOCKET_SERVER_KEY, get_socket_server
+from ..rabbitmq_settings import get_plugin_settings as get_rabbitmq_settings
+from . import _handlers
+from ._utils import (
+    APP_CLIENT_SOCKET_SERVER_KEY,
+    get_socket_server,
+    register_socketio_handlers,
+)
 
 _logger = logging.getLogger(__name__)
 
 
-async def _socketio_server_cleanup_ctx(_app: web.Application) -> AsyncIterator[None]:
+async def _socketio_server_cleanup_ctx(app: web.Application) -> AsyncIterator[None]:
+    # SEE https://github.com/miguelgrinberg/python-socketio/blob/v4.6.1/docs/server.rst#aiohttp
+    server_manager = AsyncAioPikaManager(
+        url=get_rabbitmq_settings(app).dsn, logger=_logger
+    )
+    sio_server = AsyncServer(
+        async_mode="aiohttp",
+        logger=_logger,  # type: ignore
+        engineio_logger=False,
+        client_manager=server_manager,
+    )
+    sio_server.attach(app)
+
+    app[APP_CLIENT_SOCKET_SERVER_KEY] = sio_server
+
+    register_socketio_handlers(app, _handlers)
     yield
-    # NOTE: this is ugly. It seems though that python-enginio does not
+
+    # NOTE: this is ugly. It seems though that python-socketio does not
     # cleanup its background tasks properly.
     # https://github.com/miguelgrinberg/python-socketio/discussions/1092
-    current_tasks = asyncio.tasks.all_tasks()
     cancelled_tasks = []
+
+    if hasattr(server_manager, "thread"):
+        server_thread = server_manager.thread
+        assert isinstance(server_thread, asyncio.Task)  # nosec
+        server_thread.cancel()
+        cancelled_tasks.append(server_thread)
+    if server_manager.publisher_channel:
+        await server_manager.publisher_channel.close()
+    if server_manager.publisher_connection:
+        await server_manager.publisher_connection.close()
+    current_tasks = asyncio.tasks.all_tasks()
+
     for task in current_tasks:
         coro = task.get_coro()
         if any(
             coro_name in coro.__qualname__  # type: ignore
             for coro_name in [
                 "AsyncServer._service_task",
                 "AsyncSocket.schedule_ping",
+                "AsyncPubSubManager._thread",
```

Comment:
THOUGHT: if socketiio decides to change these function names in a version, we will have zombie tasks ... we should assert or test that `AsyncServer._service_task` etc exits

Reply:
we have a test actually... but I'll maybe transform the warnings into errors..